File: backend/app/schemas/job.py
from datetime import date, datetime
from typing import Any

from pydantic import BaseModel, Field


class JobOut(BaseModel):
    id: str
    slug: str
    title: str
    dept: str | None = None
    category: str | None = None
    state_codes: list[str] = Field(default_factory=list)
    vacancies: int = 0
    qualification: str | None = None
    salary: str | None = None
    age_limit: str | None = None
    last_date: date | None = None
    apply_url: str | None = None
    pdf_url: str | None = None
    status: str = "live"
    published_at: datetime | None = None
    detail: dict[str, Any] = Field(default_factory=dict)


class JobListResponse(BaseModel):
    items: list[JobOut]
    total: int
    limit: int
    offset: int

File: frontend/src/types/job.ts
/** Shared job shapes used across cards, detail, and adapters. */

export type JobDetailBlob = {
  pdfUrls?: string[]
  pdf_urls?: string[]
  pdfUrl?: string
  pdf_url?: string
  summary?: string
  source?: string
}

export type JobRecord = {
  id?: string
  slug?: string
  title?: string
  dept?: string
  category?: string
  state?: string
  stateIds?: string[]
  vacancies?: number
  rawVacancies?: number
  qual?: string
  salary?: string
  lastDate?: string
  last_date?: string
  publishedDate?: string
  published_at?: string
  apply_url?: string
  pdfUrl?: string
  pdfUrls?: string[]
  status?: string
  detail?: JobDetailBlob
  _enriched?: boolean
  _fromLive?: boolean
  [key: string]: unknown
}
